lemur/certificates/cli.py

```diff
@@ -280,46 +289,60 @@ def rotate(endpoint_name, source, new_certificate_name, old_certificate_name, me
+            else:
+                request_rotation(endpoint, old_cert, new_cert, message, commit)
             current_app.logger.info(log_data)
 
         elif old_cert and new_cert:
             print(f"[+] Rotating all endpoints from {old_cert.name} to {new_cert.name}")
-            log_data["certificate"] = new_cert.name
-            log_data["certificate_old"] = old_cert.name
+            log_data["new_certificate"] = new_cert.name
+            log_data["old_certificate"] = old_cert.name
             log_data["message"] = "Rotating endpoint from old to new cert"
-            for endpoint in old_cert.endpoints:
-                if endpoint.primary_certificate != old_cert:
-                    # TODO(EDGE-1365) Support rotating SNI certificates.
-                    continue
-                print(f"[+] Rotating {endpoint.name}")
-                log_data["endpoint"] = endpoint.dnsname
-                request_rotation(endpoint, new_cert, message, commit)
+            rotation_eligible_endpoints = [ep for ep in old_cert.endpoints]
+            for ep in rotation_eligible_endpoints:
+                print(f"[+] Rotating {ep.name}")
+                log_data["endpoint"] = ep.dnsname
+                if region and region not in endpoint.dnsname:
+                    log_data["message"] = "Skipping rotation, region mismatch"
+                else:
+                    request_rotation(ep, old_cert, new_cert, message, commit)
                 current_app.logger.info(log_data)
-
         else:
             # No certificate name or endpoint is provided. We will now fetch all endpoints,
             # which are associated with a certificate that has been replaced
             print("[+] Rotating all endpoints that have new certificates available")
             for endpoint in endpoint_service.get_all_pending_rotation():
-                if not endpoint.certificate.replaced:
-                    # TODO(EDGE-1365) Support rotating SNI certificates.
-                    continue
-                log_data["message"] = "Rotating endpoint from old to new cert"
-                if len(endpoint.certificate.replaced) > 1:
-                    log_data["message"] = f"Multiple replacement certificates found, going with the first one out of " \
-                                          f"{len(endpoint.certificate.replaced)}"
-
-                log_data["endpoint"] = endpoint.dnsname
-                log_data["certificate"] = endpoint.certificate.replaced[0].name
-                print(
-                    f"[+] Rotating {endpoint.name} to {endpoint.certificate.replaced[0].name}"
-                )
-                request_rotation(endpoint, endpoint.certificate.replaced[0], message, commit)
-                current_app.logger.info(log_data)
+                for certificate in endpoint.certificates:
+                    log_data["message"] = "Rotating endpoint from old to new cert"
+                    if len(certificate.replaced) == 0:
```

Comment:
nit: why change this from "if not endpoint.certificate.replaced"? the original looks more "pythonic"